sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala

```diff
@@ -467,4 +468,37 @@ class DataStreamReaderWriterSuite extends StreamTest with BeforeAndAfter {
     val sq = df.writeStream.format("console").start()
     sq.stop()
   }
+
+  test("MemorySink can recover from a checkpoint in Complete Mode") {
+    val df = spark.readStream
+      .format("org.apache.spark.sql.streaming.test")
+      .load("/test")
+    val checkpointLoc = newMetadataDir
+    val checkpointDir = new File(checkpointLoc, "offsets")
+    checkpointDir.mkdirs()
+    assert(checkpointDir.exists())
+    // no exception here
+    val q = df
+      .groupBy("a")
+      .count()
+      .writeStream
+      .format("memory")
+      .queryName("test")
+      .option("checkpointLocation", checkpointLoc)
+      .outputMode("complete")
+      .start()
+
```

Comment:
This test should also check that the memory sink data is recovered to be the same as before failure.